### DISCUSSION TITLE
[Request for Beta Testers] Destinations V2 for Redshift is open for early access!

### DISCUSSION THREAD

**Discussion by cynthiaxyin** [airbytehq/airbyte#33610] [Connector Ideas and Features]
### Early Access Available!

Hi Airbyte community,

Early access for [Destinations V2](https://docs.airbyte.com/using-airbyte/core-concepts/typing-deduping) for Redshift is now available, bringing you:

- _One-to-one mapping_: Data from one stream (endpoint or table) will now create one table in the destination - gone are the days where you will need to join multiple tables to learn about your data.
- _Improved error handling_: Typing errors will no longer fail your sync, and are now easily visible in a new `_airbyte_meta` column, allowing for better tracking of inconsistencies and resolution of issues.
- _Incremental data delivery in final tables_: Data will be incrementally delivered in final tables for all initial snapshots. No more waiting hours to see the first rows in your destination table. 

### How to Enroll

We are actively seeking and iterating on feedback to refine these features prior to general availability. Here's how you can try out the beta:

1. [For Airbyte Self-Managed only] In your Workspace Settings, upgrade destination Redshift to the early access version, 0.7.4. (Airbyte Cloud will automatically upgrade. See the [changelog](https://docs.airbyte.com/integrations/destinations/redshift/#changelog) for further version updates.)

<img width="1373" alt="KEEP Screenshot 2023-12-15 at 1 11 37 PM" src="https://github.com/airbytehq/airbyte/assets/22702693/04ee6d8c-d51e-4aa4-b385-30591599dfd1">

2. Create a _new_ Redshift destination, and enable the Destinations V2 option.
**Keep in mind that the early access version is NOT suitable for production and we may ask you to reset your data.**

<img width="680" alt="KEEP Screenshot 2023-11-30 at 3 27 44 PM" src="https://github.com/airbytehq/airbyte/assets/22702693/222e29f2-225a-41a4-bf8c-ac574a03e141">

3. Create connections sending data to the new Redshift destination. These will automatically use Destinations V2.
**Please use a _different namespace_ for these connections to avoid overwriting/modifying any production data.**

4. When creating syncs with your new redshift destination, be sure to choose `Raw data (JSON)` as your Normalization Method.  In the V2 version of the Redshift destination this option will be removed, but in this "hybrid" version of the destination, this option remains for the V1 versions of the sync.  You will still be getting final tables with V2 syncs, but the mechanism we will use to generate them is no longer dbt normalization - it's built into the connector. 

### Request for Feedback

As you explore the new functionality, we ask that you pay special attention to table and column name changes, missing or incorrect column types, and the speed and cost of your syncs.

We look forward to hearing about your experience with Redshift Destinations V2. Please don't hesitate to share any feedback or questions in the comments below. You can also [join us on Slack](https://join.slack.com/share/enQtNjQzMTY5NTU1Mzg5MC1iMWJkNzJmODExOTFkNWZiNTk4OThiODRiZjI3NmRiMzAxMzY5ZGMyZmI1ODZkZmI1ZTE0NWJjYjljZTM4NDg5) if you would like to chat directly with the team about this feature.

**Comment by 4vuedemoguy** [comment]
I was tried for Redshift V2 destination on OSS setup. I got the error. Please can you verify once. I have attached Logs also.

![Screenshot 2024-01-08 at 7 51 58 PM](https://github.com/airbytehq/airbyte/assets/153283196/32fdeda6-ae83-4511-bd3c-fca3c62a8a11)

![Screenshot 2024-01-08 at 7 51 32 PM](https://github.com/airbytehq/airbyte/assets/153283196/995cca85-85f8-4f97-9fd4-fb6b85ff703d)

[default_workspace_logs_702_txt.txt](https://github.com/airbytehq/airbyte/files/13868830/default_workspace_logs_702_txt.txt)

**Reply by cynthiaxyin** [reply]
@4vuedemoguy Please make sure you follow step 4 in the original post and select `Raw data (JSON)` under the `Transformation` tab for your connection. This will make sure only V2 logic runs; from your screenshot and log file, it looks like the old V1 dbt normalization still ran.

[a comment by an automated account is not shown]
